lib/policy_machine_storage_adapters/active_record.rb

```diff
@@ -95,6 +95,161 @@ def self.persist_buffers!
       true #TODO: More useful return value?
     end
 
+    # Check conflicts for Configuration Type Role Updates
+    def self.conflicts_for_operables(operable_ids, crs_ids)
+      descendants_query = <<-SQL
+        WITH RECURSIVE descendants AS (
+          SELECT parent_id, child_id, parent_id as root_node FROM assignments
+          WHERE parent_id IN (#{operable_ids.join(", ")})
+          UNION ALL
+          SELECT a.parent_id, a.child_id, root_node FROM assignments a
```

Comment:
For the purposes of future maintainability, it may be better to alias things semantically where possible.